### PR TITLE
Adding default priors for Binomial/Bernoulli families with logit link

### PR DESCRIPTION
**Pull request by julianlheureux**
Hello! I am working on improving default priors. 

Starting with some small steps, I did some examples of logistic regression models with logit link. Linear predictor for weakly informative priors should be within -5 and 5. So I tried to improve the prior scaler, with some conditionals looking for these cases. 

Setting the intercept prior to N(0,1.5), and common terms to N(0, 1/sd(x_i)). When using categoric variables, priors will be set to N(0,1). 

Doing these changes, we arrive to weakly informative priors (at least, better than the originals).

### CODE REVIEW

**Comment by tomicapretto** [comment]
Hi @julianlheureux, thanks a lot for the contribution and sorry for the delay in the review.

Could you also re-run a few of the examples using the affected families so the docs are also updated? With that, I think we could merge this.

File: bambi/priors/scaler.py [hunk 1]
```diff
@@ -58,23 +65,66 @@ def scale_response(self):
...
 
-        # It can be greater than 1 for categorical variables
         if term.data.ndim == 1:
             mu = 0
-            sigma = self.get_slope_sigma(term.data)
+            if (
+                isinstance(self.model.family, (Bernoulli, Binomial))
+                and self.model.family.link["p"].name == "logit"
+            ):
+                if term.kind == "interaction":
```
Comment (tomicapretto) [inline bambi/priors/scaler.py:89]: Can we add a couple of comments saying what we are checking here? 

For example, whether it's a single categorical variable, an interaction between categoricals (which we treat as a single categorical for this purpose), an interaction between numeric(s) and categorical(s), or just a numeric or interaction between numerics (which we treat as a single numeric for this purpose).

**Comment by julianlheureux** [comment]
Hello! I think I successfully made the changes you highlighted. 
Please let me know if there's anything else in the code that needs improvement.

File: bambi/priors/scaler.py [hunk 2]
```diff
@@ -58,23 +65,73 @@ def scale_response(self):
     def scale_intercept(self, term):
         if term.prior.name != "Normal":
             return
-        mu, sigma = self.get_intercept_stats()
+        # Special case for logit links with bernoulli family
```
Comment (tomicapretto) [inline bambi/priors/scaler.py:68]:
```suggestion
        # Special case for logit links with bernoulli or binomial family
```

File: bambi/priors/scaler.py [hunk 3]
```diff
@@ -58,23 +65,73 @@ def scale_response(self):
...
 
     def scale_common(self, term):
         if term.prior.name != "Normal":
             return
 
-        # It can be greater than 1 for categorical variables
         if term.data.ndim == 1:
             mu = 0
-            sigma = self.get_slope_sigma(term.data)
+            # Special case for logit links with bernoulli family
```
Comment (tomicapretto) [inline bambi/priors/scaler.py:85]:
```suggestion
            # Special case for logit links with bernoulli or binomial family
```

File: bambi/priors/scaler.py [hunk 4]
```diff
@@ -58,23 +65,73 @@ def scale_response(self):
...
+                        sigma = 1
+                    else:
+                        sigma = 1 / np.std(term.data, axis=0)
+                # Single categorical term
+                elif term.categorical:
+                    sigma = 1
+                # Single numerical term
+                else:
+                    sigma = 1 / np.std(term.data, axis=0)
+            else:
```
Comment (tomicapretto) [inline bambi/priors/scaler.py:105]:
```suggestion
            # If not, fall back to the regular case
            else:
```

File: bambi/priors/scaler.py [hunk 5]
```diff
@@ -58,23 +65,73 @@ def scale_response(self):
...
+                sigma = self.get_slope_sigma(term.data)
         else:
```
Comment (tomicapretto) [inline bambi/priors/scaler.py:107]:
```suggestion
        # It's a term that spans multiple columns of the design matrix
        else:
```

File: bambi/priors/scaler.py [hunk 6]
```diff
@@ -58,23 +65,73 @@ def scale_response(self):
...
             mu = np.zeros(term.data.shape[1])
             sigma = np.zeros(term.data.shape[1])
-            # Iterate over columns in the data
-            for i, value in enumerate(term.data.T):
-                sigma[i] = self.get_slope_sigma(value)
+            # Special case for logit links with bernoulli family
```
Comment (tomicapretto) [inline bambi/priors/scaler.py:110]:
```suggestion
            # Special case for logit links with bernoulli or binomial family
```

File: bambi/priors/scaler.py [hunk 7]
```diff
@@ -58,23 +65,73 @@ def scale_response(self):
...
+                # Iterate over columns in the data
+                for i, value in enumerate(term.data.T):
+                    if term.kind == "interaction":
+                        # Distinguish cases where all interaction factor terms are categorical
+                        all_categoric = all(
+                            component.kind == "categoric" for component in term.term.components
+                        )
+                        if all_categoric:
+                            sigma[i] = 1
+                        else:
```
Comment (tomicapretto) [inline bambi/priors/scaler.py:124]:
```suggestion
                        # It's the standard deviation of the marginal numerical variable (_not_ by group)
                        else:
```

**Review by tomicapretto — COMMENTED** [review]
Hi Julián, thanks for changes. I just suggested a few extra comments.

**Review by julianlheureux — COMMENTED** [review]
I think the comments are appropriate! Thanks for helping make the code clearer and cleaner!

But there is a problem with the comments. There are some tests that fail becouse the line with the comment is too long. We have to make this shorter...

**Comment by tomicapretto** [comment]
@julianlheureux I see everything green now, so I assume you fixed the issue related to the comment length, right?

**Comment by julianlheureux** [comment]
> @julianlheureux I see everything green now, so I assume you fixed the issue related to the comment length, right?

I think i managed to fix it. If you want, you can check the changes in the last commits I made, making those lines shorter.

**Pull request merged**